flocker_bb/builders/flocker.py

```diff
@@ -73,8 +73,21 @@ def locks(self):
         # We use a shared dictionary here, since locks are compared via
         # identity, not by name.
         lock_name = '/'.join(['functional', 'api', self.provider])
+
+        # Allow up to 2 AWS functional storage driver tests to run in parallel.
+        # This is a temporary fix to get around test wait time being
+        # queued up to run.
+        # OpenStack tests have not experienced long queued wait times.
+        # So, leave the max count at 1.
+        if self.provider == 'aws':
+            maxCount = 2
+        elif self.provider in ('rackspace', 'redhat-openstack'):
+            maxCount = 1
+        else:
+            raise NotImplementedError("Unsupported provider %s" %
+                                      (self.provider,))
```

Comment:
It isn't clear that that change would be better, since there can be multiple `StorageConfiguration`s that point to the same provider.